Framework/DataObjects/src/EventWorkspace.cpp

```diff
@@ -580,6 +580,21 @@ void EventWorkspace::setAllX(const HistogramData::BinEdges &x) {
   this->clearMRU();
 }
 
+/**
+ * Set all X Histograms to a single bin with boundaries that fit the TOF of
+ * all events.
+ */
+void EventWorkspace::updateAllX() {
+  double tofmin, tofmax;
+  getEventXMinMax(tofmin, tofmax);
+
+  // Sanitize TOF min/max to ensure it always passes HistogramX validation.
+  // They would be invalid when number of events are 0 or 1, for example.
+  tofmin = tofmin < tofmax ? tofmin : 0;
+  tofmax = std::max(tofmax, tofmin + DBL_MIN);
```

Comment:
Should `std::numeric_limits<double>::min()` be used, or is `DBL_MIN` preferred in Mantid?

Reply:
This validation seems sensible to me.

Reply:
`std::numeric_limits<double>::min()` would be preferable.

Reply:
I also agree with Matthew that it would be nice to indicate that this will completely disregard the binning that is currently set and reset it to a single bin.  Something like `.resetAllXToSingleBin()`?

Reply:
`numeric_limits<double>::min()` is the smallest (largest?) negative number a double can represent, `DBL_MIN` is the smallest positive normal double.

So this is meant ensure that: `tofmin < tofmax` and `tofmax >= tofmin + DBL_MIN`.

If `tofmin = <double>::max()` and `tofmax = <double>::min()` (which `getEventXMinMax` will produce if there are no events), this will force them to `0` and `DBL_MIN` respectively.

If `tofmin = tofmax` (which `getEventXMinMax` will produce if there is exactly one event), this will leave `tofmin` alone by add `DBL_MIN` to `tofmax` to satisfy the distance requirement.

Should I make the comment in the code more detailed to explain this better?

Reply:
I think `numeric_limits<double>::lowest()` is `-1*numeric_limits<double>::max()`
`numeric_limits<double>::min()` should be equivalent to `DBL_MIN`

Reply:
You're right, my bad!